src/nxt_listen_socket.c

```diff
@@ -121,15 +121,35 @@ nxt_listen_socket_create(nxt_task_t *task, nxt_mp_t *mp,
 #endif
 
     if (nxt_socket_bind(task, s, sa) != NXT_OK) {
+
+#if (NXT_HAVE_UNIX_DOMAIN)
```

Comment:
Why bind again?

Reply:
Thus I understood the solution suggested by @VBart in the comment: https://github.com/nginx/unit/pull/649#issuecomment-1044298169.

It would be ideal to delete the socket's associated file when the program closes, but this cleanup may not occur if the container crashes or the Unit soft-terminates unsuccessfully.

Reply:
But the `sa` points to a temp file and it has been unlinked before binding.
```
        p = nxt_cpystr((u_char *) sa->u.sockaddr_un.sun_path,
                       (u_char *) orig_sa->u.sockaddr_un.sun_path);
        nxt_memcpy(p, ".tmp", 4);

        nxt_sockaddr_text(sa);

        (void) unlink(sa->u.sockaddr_un.sun_path);
```

And in the future, we need to solve the second problem mentioned in the issue. We have to clean up all the sock files during re-config. So I suggest not doing this, it introduced the logic that is not clear now.

Reply:
Thank you for bringing this point to my attention. I just saw this unlink. I apologize for not paying attention to this right away.